src/backend/distributed/utils/maintenanced.c

```diff
@@ -356,6 +361,29 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		}
 #endif
 
+		if (TimestampDifferenceExceeds(lastMetadataSyncTime,
+									   GetCurrentTimestamp(),
+									   METADATA_SYNC_TIMEOUT) && false)
+		{
+			InvalidateMetadataSystemCache();
```

Comment:
CAn we check if any nodes need sync before invalidation?

Reply:
Note that in `maintenanced.c` we don't get cache invalidation messages, so to have the most up-to-date view of `pg_dist_node`, etc. we need to invalidate the metadata cache. Otherwise we cannot check if any nodes need sync.